### ISSUE TITLE
Support PATH_TO_IMAGES in vanilla Galaxy tools.

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/galaxy#1676]
Currently only works for tool shed installed tools, differing behavior for tool shed installed tools and vanilla tools drives me crazy :smile:.

https://twitter.com/dpryan79/status/695254422956425216

**Comment by erasche** [comment]
:+1:

**Comment by nsoranzo** [comment]
:+1:

**Comment by pvanheus** [comment]
Similar request from here: https://biostar.usegalaxy.org/p/18371/ - as I proposed, possible an API call could provide a link to images/ in the tool dir?

**Comment by pjbriggs** [comment]
I don't think this issue has anything to do with the use of `$PATH_TO_IMAGES` (which is not actually required anyway, see e.g. https://github.com/galaxy-iuc/standards/pull/46) - it's more generally about how Galaxy handles the `.. image::` directive in the tool help.

As far as I can make out, Galaxy only seems to process the directive for tools which have been installed via a toolshed - specifically in the `__inititalize_help` method of the `Tool` class in `lib/galaxy/tools/__init__.py`:

        ...
        help_text = tool_source.parse_help()
        if help_text is not None:
            if self.repository_id and help_text.find( '.. image:: ' ) >= 0:
        ...

If I've understood this correctly then there would need to be some new code to handle when the tool is not an installed repository, coupled with some mechanism for actually serving the image file from wherever it is on the filesystem (as pointed out by @pvanheus). Could one possibility be to inline the images (e.g. base64 encode them)?

Either way I'd suggest change the title of this issue to remove the reference to `$PATH_TO_IMAGES`.

**Comment by SamGG** [comment]
Hi,
I am facing the same problem. What is the solution/workaround?
Best.

**Comment by SamGG** [comment]
My current workaround is to create a directory in /static/images/tools and copy the images there.
In the help section I can write:
`.. image:: ${static_path}/images/tools/my_super_tool/my_workflow_image.png`
And the image is rendered.

**Comment by pjbriggs** [comment]
Hello @SamGG it's been a while since I had this problem but I think I used the same workaround as you, if this helps.

**Comment by SamGG** [comment]
Thanks for sharing. Hope this will be fixed.
